### PR TITLE
chore: speed up build by removing bzip and libz from critical path

### PR DESCRIPTION
**Pull request by hanabi1224**
**Summary of changes**
Changes introduced in this pull request:
- Speed up build by removing `bzip` and `libz` from critical path
- Try CI without buildjet
- `libz` and `bzip` features of `rocksdb` are disabled
- `libp2p-websoket` is removed. Alternatively we can patch the `soketto` or `flate2` crate to get rid of `libz-sys` (`flate2` uses rust implement of the compression algorithm by default, for some reason `soketto` specifies the `libz` feature)
- `libz` and `bzip` remains in `Cargo.lock` maybe because `librocksdb-sys` has `libz-sys?/static` feature.

Given the number, I think we can use gh hosted runner for calibnet sync check job, given MacOS build is likely to be slower

Build time  on hosted runner

|job|[cache miss](https://github.com/ChainSafe/forest/actions/runs/3495118179/jobs/5851529367)|[cache hit](https://github.com/ChainSafe/forest/actions/runs/3495411368)|migrate?|
|--|--|--|--|
|test|42m10s|15m29s| no |
|calibnet sync check|25m37s|9m41s| yes? |

For reference, [test job on buildjet with cache](https://github.com/ChainSafe/forest/actions/runs/3493035550) takes `5m11s`. (This job does not have `bzip` and `libz` removed)

Build time report of `main` from CI build artifacts:

| Rank |Unit | Total | Codegen | Features|
| -- | -- | -- | -- | -- |
|1. | wasmtime-runtime v1.0.2 build script (run) | 63.5s |   | pooling-allocator|
|2. | libz-sys v1.1.8 build script (run) | 59.9s |   | static
|3. | psm v0.1.21 build script (run) | 59.4s |   |  
|4. | sha2-asm v0.6.2 build script (run) | 58.0s |   |  
|5. | blake3 v1.3.1 build script (run) | 54.2s |   |  
|6. | bzip2-sys v0.1.11+1.0.8 build script (run) | 36.4s |   | static
|7. | librocksdb-sys v0.8.0+7.4.4 build script (run) | 34.0s |   | bzip2, bzip2-sys, default, libz-sys, lz4, snappy, static, zlib
|8. | blst v0.3.9 build script (run) | 29.0s |   | default
|9. | ring v0.16.20 build script (run) | 23.6s |   | alloc, default, dev_urandom_fallback, once_cell, std
|10. | proc-macro2 v1.0.47 build script | 8.2s |   | default, proc-macro, span-locations

Clean build time report on my laptop (no `target` dir, no sccache), with `cargo check --timings` 
full report: [build-time.zip](https://github.com/ChainSafe/forest/files/10043154/build-time.zip)

TLDR:
|branch|total time|ratio|
|--|--|--|
|main|212.2s|100%|
|pr|184.8s|87.09%|

**Reference issue to close (if applicable)**

Closes part of https://github.com/ChainSafe/forest/issues/2088

**Other information and links**

### CODE REVIEW

File: node/forest_libp2p/src/service.rs [hunk 1]
```diff
@@ -782,8 +781,8 @@ async fn emit_event(sender: &flume::Sender<NetworkEvent>, event: NetworkEvent) {
 pub async fn build_transport(local_key: Keypair) -> Boxed<(PeerId, StreamMuxerBox)> {
     let tcp_transport =
         || libp2p::tcp::TokioTcpTransport::new(libp2p::tcp::GenTcpConfig::new().nodelay(true));
-    let transport = libp2p::websocket::WsConfig::new(tcp_transport()).or_transport(tcp_transport());
-    let transport = libp2p::dns::TokioDnsConfig::system(transport).unwrap();
+    // let transport = libp2p::websocket::WsConfig::new(tcp_transport()).or_transport(tcp_transport());
```
Comment (LesnyRumcajs) [inline node/forest_libp2p/src/service.rs:784]:
```suggestion
```

**Review by LesnyRumcajs — APPROVED** [review]
Great!

**Review by lemmih — APPROVED** [review]
